src/stdlib/match_datadog_query.rs

```diff
@@ -86,6 +88,38 @@ impl Function for MatchDatadogQuery {
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum Error {
+    InvalidFilter(PathParseError),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::InvalidFilter(err) => err.fmt(f),
+        }
+    }
+}
+
+impl std::error::Error for Error {}
+
+impl DiagnosticMessage for Error {
+    fn code(&self) -> usize {
+        112
```

Comment:
We could consider reusing the 403 error code ("invalid argument") for this: https://github.com/vectordotdev/vector/blob/master/website/cue/reference/remap/errors/403_invalid_argument.cue

Reply:
Done in https://github.com/vectordotdev/vrl/pull/1031/commits/6d01712ed53438a08464fed783734b4e61746ed0.